uaclient/security.py

```diff
@@ -913,6 +918,52 @@ def _check_subscription_for_required_service(
     return False
 
 
+def _prompt_for_new_token(cfg: UAConfig) -> bool:
+    """Prompt for attach a new subscription token to the user.
+
+    :return: True if attach performed.
+    """
+    import argparse
+    from uaclient import cli
+
+    _check_cloud_type()
+    print(status.MESSAGE_SECURITY_UPDATE_NOT_INSTALLED_EXPIRED)
+    token_url = "https://ubuntu.com/advantage"
+    choice = util.prompt_choices(
+        "Choose: [R]enew your subscription (at {}) [C]ancel".format(token_url),
+        valid_choices=["r", "c"],
+    )
+    if choice == "r":
+        print(status.PROMPT_EXPIRED_ENTER_TOKEN)
+        token = input("> ")
+        print(status.colorize_commands([["ua", "detach"]]))
+        cli.action_detach(argparse.Namespace(assume_yes=True), cfg)
+
+        print(status.colorize_commands([["ua", "attach", token]]))
+        return bool(
+            0
+            == cli.action_attach(
+                argparse.Namespace(token=token, auto_enable=True), cfg
+            )
+        )
+
+    return False
+
+
+def _check_subscription_is_expired(cfg: UAConfig) -> bool:
+    from uaclient import cli
+
+    try:
```

Comment:
Can we emit colorize commands for `ua refresh` here to represent the changes being performed on the system.
I filed a separate issue to make sure we do this in general for ua fix work https://github.com/canonical/ubuntu-advantage-client/issues/1477

Reply:
I though about doing that, however, I don't think we should surface that to the user. We are only running that command to check if the subscription is not expired, meaning that most of the times, it will probably not change the system at all.

But you think we can still output it, no problem

Reply:
I'm on the fence, but not fully convinced we need it, ua refresh does make a change if there are contract changes in general like changes in aptURL, packages required etc.  The only exposure I'd think about currently  is Ubuntu PRO images, once we release 26.2 to AWS/GCP/Azure, we will lock-step update contracts changing additionalPackages directives. I guess if a PRO image updated to 27.0 we might have a case where running a ua fix before reboot would attempt to apply those contract deltas and break on apt installs of ubuntu-fips metapkg due to current deb pinning. I suppose this is a small  corner-case and maybe negligible here.

Reply:
I think this is good point. Maybe we could have another way to check if the subscription is expired or not instead of running refresh. We could see if the contracts have an endpoint that allows us to do just that.